test/performance/io/format_sam_benchmark.cpp

```diff
@@ -0,0 +1,331 @@
+#include <type_traits>
+#include <streambuf>
+#include <seqan3/alignment/pairwise/alignment_result.hpp>
+#include <seqan3/alignment/pairwise/alignment_configurator.hpp>
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/alphabet/nucleotide/dna15.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/alphabet/detail/alphabet_base.hpp>
+#include <seqan3/io/alignment_file/output.hpp>
+#include <seqan3/io/alignment_file/input.hpp>
+#include <seqan3/io/alignment_file/format_sam.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/bam_io.h>
+#endif
+
+using namespace seqan3;
+
+constexpr auto default_seed = 1234u;
+size_t const reference_length = 50;
+
+// ============================================================================
+// generate mutated sequence from reference
+// ============================================================================
+enum class Mutation_type : u_int8_t
+{
+    DELETION,
+    INSERTION,
+    SUBSTITUTION
+};
+
+float const mutation_probability = 0.1;
+
+template<Alphabet alphabet_t>
+std::vector<alphabet_t> mutate_sequence(std::vector<alphabet_t> input_sequence,
+                                        size_t seed = default_seed,
+                                        float mutation_probability = mutation_probability)
+{
+    std::mt19937 rng_engine(seed);
+
+    // c.f. for loop below
+    std::uniform_real_distribution<float> cointoss_distribution(0, 1);
+    std::uniform_int_distribution<u_int8_t> mutation_distribution(0, 3 -1); // 3 = n values in Mutation_type;
+    std::uniform_int_distribution<u_int8_t> base_distribution(0, alphabet_size<alphabet_t> -1);
+
+    std::vector<alphabet_t> mutated_sequence{};
+
+    for (std::size_t i = 0; i < input_sequence.size(); ++i)
+    {
+        // cointoss if letter should be mutated at all
+        if (cointoss_distribution(rng_engine) > mutation_probability)
+        {
+            mutated_sequence.push_back(input_sequence[i]);
+            continue;
+        }
+        else
+        {
+            // determine type of mutation and mutate
+            Mutation_type mutation_type = static_cast<Mutation_type>(mutation_distribution(rng_engine));
+
+            switch (mutation_type)
+            {
+                case Mutation_type::DELETION:
+                    break; // skip
+
+                case Mutation_type::INSERTION :
+                    mutated_sequence.push_back(alphabet_t{}.assign_rank(mutation_distribution(rng_engine)));
+                    mutated_sequence.push_back(input_sequence[i]);
+
+                    break;
+
+                case Mutation_type::SUBSTITUTION:
+                    mutated_sequence.push_back(alphabet_t{}.assign_rank(mutation_distribution(rng_engine)));
+                    break;
+            }
+        }
+    }
+
+    return mutated_sequence;
+}
```

Comment:
I wanted to ask whether I should move this function into it's own header close to ``test/performance/sequence_generator.hpp`` since it's technically generic enough to be reused for test cases later on